pkg/config/config.go

```diff
@@ -155,3 +155,21 @@ func LoadConfFromCache(args *skel.CmdArgs) (*sriovtypes.NetConf, string, error)
 
 	return netConf, cRefPath, nil
 }
+
+// GetMacAddressForResult return the mac address we should report to the CNI call return object
+// if the device is on kernel mode we report that one back
+// if not we check the administrative mac address on the PF
+// if it is set and is not zero, report it.
+func GetMacAddressForResult(netConf *sriovtypes.NetConf) string {
+	if netConf.MAC != "" {
```

Comment:
do we need this statement ?

if MAC is specified then we set as admin mac and in non-dpdk we set as effective as well.
so the two statements below will cover it.

plus it will match the description of the method.

Reply:
This will not work for DPDK + static mac.
because we use `netConf.OrigVfState.AdminMAC` that contains the original adminMac not the static one we request in the `MAC` object

Reply:
I missed the fact that what we return below is the "Original" VF state so indeed when we override, the only place we keep that value is in `netconf.MAC`

sorry for the confusion.